### PR TITLE
#727 - adding sosreport role in infrastructure

### PR DESCRIPTION
**Pull request by strus38**
Hi

New feature #727 
Generate a sos_report script and check sos package dependencies for all mgmt_* nodes (computes and mggmt only).
Carefull, no filter on bmc ... so it will fail for them (which is not beautiful, but still works).

The script is gathering sos report, and also takes the roles and inventory content. More can be added if we need, but that would be the minimal to me to start with.

```
$ cat defaults/main.yml
sosreport_binary_path: /usr/local/bin
sosreport_binary_name: sosreport-bb
sosreport_output_path: /tmp/sosreport
sosreport_tempfile_path: /tmp
```

**Warning** The script is gathering information on all servers (management, login and computes).

```
$ cat playbooks/sosreport.yml
- name: Install and configure sosreport
  hosts: "mg_managements, mg_logins, mg_computes"
  roles:
    - name: sosreport

$ ansible-playbook playbooks/sosreport.yml

$ sosreport-bb
Package manager: dnf
=== Checking script dependencies
=== Checking node dependencies
=== Gathering all nodes sos reports...

sos-collector (version 4.3)
...
Beginning collection of sosreports from 22 nodes, collecting a maximum of 4 concurrently
mgmt1  : Generating sos report...
compute1002 : Generating sos report...
compute10025 : Generating sos report...
compute10006 : Generating sos report...
...
/tmp/sosreport/sos_error_logs.txt
/tmp/sosreport/sos-collector-2023-01-30-svfbp.tar.xz

######################################################################
##
## INFO - Please MANUALLY delete temporary report files: /tmp/sosreport
## Please send the /var/logs/sosreport/sosreport-2023-01-30.tgz file to Support team
## with the SHA1: 0262e1d8304ad4023c268c39e6b9213ce4de2fce  /var/logs/sosreport/sosreport-2023-01-30.tgz
##
######################################################################
```

### CODE REVIEW

File: collections/infrastructure/roles/sosreport/templates/sosreport.sh.j2 [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+tmp="{{ sosreport_tempfile_path }}"
+output_path="{{ sosreport_output_path }}"
+output_file="sosreport"
+
+function usage {
+    cat <<EOF
+Usage:
+  $0 [options]
+
+Note: the script must be run on the OMN node
```
Comment (ginomcevoy) [inline collections/infrastructure/roles/sosreport/templates/sosreport.sh.j2:29]: I would replace "the OMN node" with "a management node".

Comment (strus38) [reply collections/infrastructure/roles/sosreport/templates/sosreport.sh.j2:29]: yes indeed. Thanks.

**Comment by oxedions** [comment]
Dear @strus38 

Thanks a lot for this PR, and apologies for the delay of my answer. Tool is a good base, we can make it evolve on the time.
2 things I would like to be updated before merging:

1. Please move the role into collections/stack/roles since it is a "stack" related tool. I will improve the stack collection as soon as I have stabilized other collections.
2. Please rename the tool to bluebanquise-sosreport to match the other tools naming convention.

Then all good for me 😊

**Comment by strus38** [comment]
@oxedions - ready for review. Fixed issues found during PR review.

**Review by oxedions — APPROVED** [review]
Dear @strus38 

Apologies for the delay.
This is ok to me 😊 

Merging

**Pull request merged**